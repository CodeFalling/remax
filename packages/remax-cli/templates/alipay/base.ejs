<template name="REMAX_TPL">
  <block a:for="{{root.children}}" a:key="{{item.id}}">
    <template is="REMAX_TPL_1" data="{{item}}"/>
  </block>
</template>

<template name="REMAX_TPL_1">
  <block>
    <block a:if="{{item && item.text}}">
      <!-- string -->
      {{item.text}}
    </block>

    <% for (let component of components) { %>
      <%- include('./component.ejs', {
        type: component.type,
        props: component.props,
        id: component.id,
      }) %>
    <% } %>
  </block>
</template>
