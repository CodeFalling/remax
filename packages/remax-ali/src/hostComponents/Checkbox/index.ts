import * as React from 'react';
import { createHostComponent } from '@remax/shared';

export interface CheckboxProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  name?: string;
  className?: string;
  value?: string;
  checked?: boolean;
  disabled?: boolean;
  color?: string;
  onChange?: (e: any) => void;
}

export const Checkbox: React.ComponentType<CheckboxProps> = createHostComponent<CheckboxProps>('checkbox');
