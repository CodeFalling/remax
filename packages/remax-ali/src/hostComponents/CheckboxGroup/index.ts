import * as React from 'react';
import { createHostComponent } from '@remax/shared';

export interface CheckboxGroupProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  name?: string;
  onChange?: (e: any) => void;
}

export const CheckboxGroup: React.ComponentType<CheckboxGroupProps> = createHostComponent<CheckboxGroupProps>(
  'checkbox-group'
);
