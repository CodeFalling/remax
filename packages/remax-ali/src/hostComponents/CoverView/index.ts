import * as React from 'react';
import { createHostComponent } from '@remax/shared';

export interface CoverViewProps {
  readonly dataset?: DOMStringMap;
  className?: string;
  style?: React.CSSProperties;
  onTap?: (e: any) => void;
}

export const CoverView: React.ComponentType<CoverViewProps> = createHostComponent<CoverViewProps>('cover-view');
