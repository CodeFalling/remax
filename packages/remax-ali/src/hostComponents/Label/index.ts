import * as React from 'react';
import { createHostComponent } from '@remax/shared';

export interface LabelProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  for?: string;
  className?: string;
}

export const Label: React.ComponentType<LabelProps> = createHostComponent<LabelProps>('label');
