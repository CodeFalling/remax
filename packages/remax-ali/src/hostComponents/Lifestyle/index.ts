import * as React from 'react';
import { createHostComponent } from '@remax/shared';

export interface LifestyleProps {
  readonly dataset?: DOMStringMap;
  publicId: string;
  onFollow?: (e: any) => void;
}

export const Lifestyle: React.ComponentType<LifestyleProps> = createHostComponent<LifestyleProps>('lifestyle');
