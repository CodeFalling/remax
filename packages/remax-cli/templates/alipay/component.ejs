<% camelCased = function(str) {
  if (str.indexOf('-') !== -1) {
    return str;
  }
  return str.replace(/-([a-z0-9])/g, function (g) { return g[1].toUpperCase();
})}%>

<template name="REMAX_TPL_1_<%=id%>">
  <<%=id%> <% for(let key of props) { %> <%=key%>="{{item.props['<%=camelCased(key)%>']}}" <% } %>>
    <% if (id === 'swiper') { %>
    <block a:for="{{item.children}}" a:key="{{item.id}}">
      <swiper-item key="{{item.props.key}}">
        <block a:for="{{item.children}}" a:key="{{item.id}}">
          <template is="REMAX_TPL_1_CONTAINER" data="{{item: item}}" />
        </block>
      </swiper-item>
    </block>
    <% } else if (id === 'picker') { %>
    <view>
      <template is="REMAX_TPL_1_CONTAINER" data="{{item: item.children[0]}}" />
    </view>
    <% } else if (id === 'picker-view') { %>
    <block a:for="{{item.children}}" a:key="{{item.id}}">
      <picker-view-column>
        <view a:for="{{item.children}}">
          <template is="REMAX_TPL_1_CONTAINER" data="{{item}}" />
        </view>
      </picker-view-column>
    </block>
    <% } else { %>
    <block a:for="{{item.children}}" a:key="{{item.id}}">
      <template is="REMAX_TPL_1_CONTAINER" data="{{item: item}}" />
    </block>
    <% } %>
  </<%=id%>>
</template>