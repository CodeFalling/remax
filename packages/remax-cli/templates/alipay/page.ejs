<import-sjs name="helper" from="<%= jsHelper %>"/>
<import src="<%= baseTemplate %>"/>
<template is="REMAX_TPL" data="{{tree: helper.reduce(action)}}"/>
