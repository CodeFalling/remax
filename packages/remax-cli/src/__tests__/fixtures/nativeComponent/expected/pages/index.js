import '../npm/remax/esm/render.js';
import { createRef, createElement } from 'react';
import '../npm/remax/esm/createAppConfig.js';
import '../npm/remax/esm/Platform.js';
import '../npm/remax/esm/createHostComponent.js';
import createPageConfig from '../npm/remax/esm/createPageConfig.js';
import createNativeComponent from '../npm/remax/esm/createNativeComponent.js';
import '../npm/remax/esm/index.js';
import View from '../npm/remax/esm/adapters/alipay/components/View.js';
import '../npm/remax/esm/adapters/alipay/components/ScrollView.js';
import '../npm/remax/esm/adapters/alipay/components/Swiper.js';
import '../npm/remax/esm/adapters/alipay/components/SwiperItem.js';
import '../npm/remax/esm/adapters/alipay/components/MovableView.js';
import '../npm/remax/esm/adapters/alipay/components/MovableArea.js';
import '../npm/remax/esm/adapters/alipay/components/CoverView.js';
import '../npm/remax/esm/adapters/alipay/components/CoverImage.js';
import '../npm/remax/esm/adapters/alipay/components/Icon.js';
import '../npm/remax/esm/adapters/alipay/components/Text.js';
import '../npm/remax/esm/adapters/alipay/components/RichText.js';
import '../npm/remax/esm/adapters/alipay/components/Progress.js';
import '../npm/remax/esm/adapters/alipay/components/Button.js';
import '../npm/remax/esm/adapters/alipay/components/CheckboxGroup.js';
import '../npm/remax/esm/adapters/alipay/components/Checkbox.js';
import '../npm/remax/esm/adapters/alipay/components/Form.js';
import '../npm/remax/esm/adapters/alipay/components/Input.js';
import '../npm/remax/esm/adapters/alipay/components/Label.js';
import '../npm/remax/esm/adapters/alipay/components/Picker.js';
import '../npm/remax/esm/adapters/alipay/components/PickerView.js';
import '../npm/remax/esm/adapters/alipay/components/PickerViewColumn.js';
import '../npm/remax/esm/adapters/alipay/components/RadioGroup.js';
import '../npm/remax/esm/adapters/alipay/components/Radio.js';
import '../npm/remax/esm/adapters/alipay/components/Slider.js';
import '../npm/remax/esm/adapters/alipay/components/Switch.js';
import '../npm/remax/esm/adapters/alipay/components/TextArea.js';
import '../npm/remax/esm/adapters/alipay/components/Navigator.js';
import '../npm/remax/esm/adapters/alipay/components/Image.js';
import '../npm/remax/esm/adapters/alipay/components/Map.js';
import '../npm/remax/esm/adapters/alipay/components/Canvas.js';
import '../npm/remax/esm/adapters/alipay/components/WebView.js';
import '../npm/remax/esm/adapters/alipay/components/Lifestyle.js';
import '../npm/remax/esm/adapters/alipay/components/ContactButton.js';
import '../npm/remax/esm/adapters/alipay/components/Video.js';
import '../npm/remax/esm/adapters/alipay/api.js';

var SrcComponent = createNativeComponent('src-0');
var ScopedComponent = createNativeComponent('a-1');
var SlotComponent = createNativeComponent('slot-0');
var PluginComponent2 = createNativeComponent('my-plugin-1');
var PluginComponent = createNativeComponent('my-plugin-0');
var Complex = createNativeComponent('complex-0');
var E = createNativeComponent('e-0');
var D = createNativeComponent('d-0');
var C = createNativeComponent('c-0');
var B = createNativeComponent('b-0');
var A = createNativeComponent('a-0');












var _page = function _page() {
  var b = createRef();
  return createElement(View, null, createElement(A, {
    foo: "bar"
  }), createElement(B, {
    ref: b
  }), createElement(C, null), createElement(D, null), createElement(E, null), createElement(Complex, null), createElement(PluginComponent, null), createElement(PluginComponent2, null), createElement(SlotComponent, null, createElement(View, {
    slot: "inner"
  })), createElement(ScopedComponent, null), createElement(SrcComponent, null));
};

var index = Page(createPageConfig(_page));

export default index;
