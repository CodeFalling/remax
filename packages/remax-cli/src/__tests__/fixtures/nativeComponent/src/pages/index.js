import * as React from 'react';
import { View } from 'remax/alipay';
import A from '../components/a';
import B from '@/components/b';
import C from '../components/c/index';
import D from '@c/d/index.js';
import E from '@components/e/index.js';
import Complex from '../components/complex';
import PluginComponent from 'plugin://myPlugin/xx';
import PluginComponent2 from 'plugin://myPlugin/dddd';
import SlotComponent from '../components/slot';
import ScopedComponent from '../components/@foo/a';
import SrcComponent from '../components/src';
import NotInJSXComponent from '../components/notInJSX';
import CJSComponent from 'cjs';

export default () => {
  const b = React.createRef();
  const text = 'not in jsx' + NotInJSXComponent;
  return (
    <View>
      <A foo="bar" />
      <B ref={b} />
      <C />
      <D />
      <E />
      <Complex />
      <PluginComponent />
      <PluginComponent2 />
      <SlotComponent>
        {/* 测试 slot view 的 prop 收集 */}
        <View id="inner" slot="inner"></View>
      </SlotComponent>
      <SlotComponent>
        {/* 测试 slot view 的 prop 收集 */}
        <View className="outer" slot="outer" ns:attr="value" />
      </SlotComponent>
      <ScopedComponent />
      <SrcComponent />
      <CJSComponent />
      {text}
    </View>
  );
};
