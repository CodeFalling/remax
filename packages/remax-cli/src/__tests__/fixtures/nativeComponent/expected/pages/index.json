{
  "usingComponents": {
    "a-0": "../components/a/index",
    "b-1": "../components/b/index",
    "c-2": "../components/c/index",
    "d-3": "../components/d/index",
    "e-4": "../components/e/index",
    "complex-5": "../components/complex/index",
    "my-plugin-6": "plugin://myPlugin/xx",
    "my-plugin-7": "plugin://myPlugin/dddd",
    "slot-8": "../components/slot/index"
  }
}